src/main/java/com/iexec/worker/executor/TaskNotificationService.java

```diff
@@ -239,10 +245,40 @@ private TaskNotificationType updateStatusAndGetNextAction(String chainTaskId, Re
         log.info("update replicate request [chainTaskId:{}, status:{}, details:{}]",
                 chainTaskId, statusUpdate.getStatus(), statusUpdate.getDetailsWithoutLogs());
 
-        TaskNotificationType next = customCoreFeignClient.updateReplicateStatus(chainTaskId, statusUpdate);
+        TaskNotificationType next = null;
+        while (next == null && !isFinalDeadlineReached(chainTaskId, Instant.now().toEpochMilli())) {
+            next = customCoreFeignClient.updateReplicateStatus(chainTaskId, statusUpdate);
+        }
 
         log.info("update replicate response [chainTaskId:{}, status:{}, next:{}]",
                 chainTaskId, statusUpdate.getStatus(), next);
         return next;
     }
-}
\ No newline at end of file
+
+    /**
+     * Checks whether the final deadline is reached.
+     * If never called before for this task, retrieves the final deadline from the chain and caches it.
+     * If already called, then retrieves the final deadline from the cache.
+     * <p>
+     * Note that if the task is unknown on the chain, then the final deadline is considered as reached.
+     *
+     * @param chainTaskId Task ID whose final deadline should be checked.
+     * @param now         Time to check final deadline against.
+     * @return {@literal true} if the final deadline is met or the task is unknown on-chain,
+     * {@literal false} otherwise.
+     */
+    boolean isFinalDeadlineReached(String chainTaskId, long now) {
+        if (finalDeadlineForTask.containsKey(chainTaskId)) {
+            return now >= finalDeadlineForTask.get(chainTaskId);
```

Comment:
Can you remove duplication for <= checks please?

Reply:
See https://github.com/iExecBlockchainComputing/iexec-worker/pull/483/commits/bd453e89997f35bb40f528255ba53ca1385cb5a8.

Reply:
Thank you :pray: